### ISSUE TITLE
When I start nakama using the docker-compose file I get the connection error. Also OnSuccess delegate is not getting fired. 

### ISSUE THREAD

**Issue by syedalamabbas** [heroiclabs/nakama-unreal#7]
I use the familiar file from: 
`wget https://raw.githubusercontent.com/heroiclabs/nakama/master/install/docker/docker-compose.yml`

and when I start and check the nakama server I get the following error in the log. 
`alam@ubuntu:~$ docker ps

CONTAINER ID        IMAGE                          COMMAND                  CREATED             STATUS                 PORTS                                              NAMES

06bc93fcf1f2        heroiclabs/nakama:latest       "/bin/bash -ecx '/..."   7 hours ago         Up 7 hours (healthy)   0.0.0.0:7350-7351->7350-7351/tcp                   nkserver_nakama_1

6842038d4f8c        cockroachdb/cockroach:latest   "/cockroach/cockro..."   7 hours ago         Up 7 hours             0.0.0.0:8080->8080/tcp, 0.0.0.0:26257->26257/tcp   nkserver_cockroachdb_1

alam@ubuntu:~$ docker exec -it nkserver_nakama_1 bash

root@06bc93fcf1f2:/nakama# ./nakama logs

{"level":"warn","ts":"2017-10-04T13:46:39.815Z","msg":"WARNING: insecure default parameter value, change this for production!","param":"socket.server_key"}
{"level":"warn","ts":"2017-10-04T13:46:39.815Z","msg":"WARNING: insecure default parameter value, change this for production!","param":"session.encryption_key"}
{"level":"warn","ts":"2017-10-04T13:46:39.816Z","msg":"WARNING: insecure default parameter value, change this for production!","param":"runtime.http_key"}
{"level":"info","ts":"2017-10-04T13:46:39.816Z","msg":"Nakama starting"}
{"level":"info","ts":"2017-10-04T13:46:39.816Z","msg":"Node","name":"nakama-a391","version":"1.0.2+708da56"}
{"level":"info","ts":"2017-10-04T13:46:39.816Z","msg":"Data directory","path":"/nakama/data"}
{"level":"info","ts":"2017-10-04T13:46:39.816Z","msg":"Database connections","dsns":["root@localhost:26257"]}
{"level":"fatal","ts":"2017-10-04T13:46:39.817Z","msg":"Error pinging database","error":"dial tcp 127.0.0.1:26257: getsockopt: connection refused","stacktrace":"nakama/vendor/go.uber.org/zap.Stack\n\tvendor/go.uber.org/zap/field.go:209\nnakama/vendor/go.uber.org/zap.(*Logger).check\n\tvendor/go.uber.org/zap/logger.go:273\nnakama/vendor/go.uber.org/zap.(*Logger).Fatal\n\tvendor/go.uber.org/zap/logger.go:208\nmain.dbConnect\n\tmain.go:157\nmain.main\n\tmain.go:81"}

root@06bc93fcf1f2:/nakama# 
`

I am trying to use the given blueprint nodes like this :  
![nakamaproblem](https://user-images.githubusercontent.com/9631714/31179711-ee7187fc-a8d9-11e7-9122-a11c0964a83a.png)

Here is the code to setup the nakama: 

`// Fill out your copyright notice in the Description page of Project Settings.
#include "NakamaGameInstance.h"
#include "Private/Blueprints/NakamaComponent.h"
#include "NakamaTestProj.h"

UNakamaGameInstance::UNakamaGameInstance(const FObjectInitializer& ObjectInitializer)
{
	MainNakamaComp = CreateDefaultSubobject<UNakamaComponent>(TEXT("MainNakamaComponent"));
	// Binding here

	onSuccessMatchRequestDelegate.BindUFunction(this, FName("SuccessfullMatchMade"));
	OnFailMatchRequestDelegate.BindUFunction(this, FName("FailedToCreateMatch"));

	onSuccessAutheticateRequestDelegate.BindUFunction(this, FName("SuccessfullyAuthenticated"));
	OnFailAutheticateRequestDelegate.BindUFunction(this, FName("FailedToAuthenticate"));
}

void UNakamaGameInstance::CallSetupNakamaClient()
{
	FString serverKey = "defaultkey";
	FString host = "192.168.71.130";
	int32 port = 7350;
	FString lang = "en";
	bool ssl = false;

	if(onSuccessAutheticateRequestDelegate.IsBound() && OnFailAutheticateRequestDelegate.IsBound())
		MainNakamaComp->SetupClient(serverKey, host, port, lang, ssl);

	//UNBPMatchRequest::CreateMatch(MainNakamaComp, onSuccessMatch, onFail);
	//Nakama::NClient someClient = Nakama::NClient::Builder("defaultkey").Host("192.168.71.130").Port(7350).SSL(false).Build();

	//std::string token = "sometoken";
	//// record start time
	//auto start = std::chrono::system_clock::now();
	//auto duration = start.time_since_epoch();
	//std::chrono::milliseconds millis = std::chrono::duration_cast<std::chrono::milliseconds>(duration);

	//Nakama::NSession* someSession = new Nakama::NSession(token, millis);
	//
	//someClient.Connect(someSession);

	//// We'll use device ID for the user. See other authentication options.
	//Nakama::NAuthenticateMessage someMessage = Nakama::NAuthenticateMessage::Custom("SimpleTest");

	//someClient.Register(someMessage, nullptr, nullptr);

	//someClient.Login(someMessage, nullptr, nullptr);

	//std::string someID = someSession->GetId();

	//FString SessionID(someID.c_str());

	//UE_LOG(NakamaTestLog, Warning, TEXT("How do we know if the sessions is started %s"), *SessionID);
	
}

void UNakamaGameInstance::SuccessfullMatchMade(UNBPMatch * match)
{
	GEngine->AddOnScreenDebugMessage(-1, 15.f, FColor::Green, TEXT("This is a success. We have successfully made a match !"));
}

void UNakamaGameInstance::FailedToCreateMatch(UNBPError * error)
{
	GEngine->AddOnScreenDebugMessage(-1, 15.f, FColor::Red, FString("This is a failure! We failed to create a match because ") + error->GetErrorMessage());
}

void UNakamaGameInstance::SuccessfullyAuthenticated()
{
	GEngine->AddOnScreenDebugMessage(-1, 15.f, FColor::Green, TEXT("This is a success. We have successfully authenticated!"));
}

void UNakamaGameInstance::FailedToAuthenticate(UNBPError * error)
{
	GEngine->AddOnScreenDebugMessage(-1, 15.f, FColor::Red, FString("This is a failure! We failed to authenticate because ") + error->GetErrorMessage());
}
`

**Comment by novabyte** [comment]
@syedalamabbas The logs show that the database server isn't reachable:

```
{"level":"fatal","ts":"2017-10-04T13:46:39.817Z","msg":"Error pinging database","error":"dial tcp 127.0.0.1:26257: getsockopt: connection refused","stacktrace":"nakama/vendor/go.uber.org/zap.Stack\n\tvendor/go.uber.org/zap/field.go:209\nnakama/vendor/go.uber.org/zap.(*Logger).check\n\tvendor/go.uber.org/zap/logger.go:273\nnakama/vendor/go.uber.org/zap.(*Logger).Fatal\n\tvendor/go.uber.org/zap/logger.go:208\nmain.dbConnect\n\tmain.go:157\nmain.main\n\tmain.go:81"}
```

Specifically "dial tcp 127.0.0.1:26257: getsockopt: connection refused". Do you actually have the database server started with the Docker compose file?

**Comment by syedalamabbas** [comment]
See the command just before the logs : 
```
alam@ubuntu:~$ docker ps

CONTAINER ID IMAGE COMMAND CREATED STATUS PORTS NAMES

06bc93fcf1f2 heroiclabs/nakama:latest "/bin/bash -ecx '/..." 7 hours ago Up 7 hours (healthy) 0.0.0.0:7350-7351->7350-7351/tcp nkserver_nakama_1

6842038d4f8c cockroachdb/cockroach:latest "/cockroach/cockro..." 7 hours ago Up 7 hours 0.0.0.0:8080->8080/tcp, 0.0.0.0:26257->26257/tcp nkserver_cockroachdb_1
```

Docker cockroach container nicely exists on its own, when I try to log in it even gives an HTTP response saying the user is already in use. I can check the dashboard, but no table entries are being created for the user with the device id.

**Comment by syedalamabbas** [comment]
I decided to check if the database tables had any entries. I found out that it does - the login with the register/login with deviceid indeed works. 

![nakamaproblem3](https://user-images.githubusercontent.com/9631714/31213238-c51faa76-a962-11e7-96df-f708ef0b3005.png)

I don't know then why there this that error in the socket connection. Is it then related to create match - since that is the last operation I do from the blueprint.  Also, the problem that the delegates don't fire on login and register blueprints nodes persist.

**Comment by talothman** [comment]
Find a solution for this? I've been trying to get Nakama running with no success.

**Comment by wm-Matt** [comment]
Any progress on this? We have encountered the same issue.

**Comment by mofirouz** [comment]
We've released Nakama-Unreal v2.0.3. I don't think this is relevant anymore and I'll close the ticket. Feel free to reopen if you encounter the same issue with the new version.
